packages/async-flow/src/async-flow.js

```diff
@@ -162,8 +166,9 @@ export const prepareAsyncFlowTools = (outerZone, outerOptions = {}) => {
            * start of a replay.
            *
            * @param {boolean} [eager]
+           * @param {boolean} [eventualSend]
            */
-          restart(eager = startEager) {
+          restart(eager = startEager, eventualSend = enableEventualSend) {
```

Comment:
I am feeling very unsure about making this a dynamic option.

The problem is that we likely cannot ever support an async flow definition changing its support for eventual send: if the guest code changes to start using eventual send, then most likely it'll fail replaying previous invocations that didn't use it. At least not until we have some sort of support for diverging replay behavior (#10103), and even then I'm suspicious we'd want to encourage such changes.

Interestingly that also means that even when we flip the switch to support eventual send by default, we might want to continue disabling support in existing flow definitions. So far we've always thought we can enable new features that previously were errors, but the fact that async-flows can have both previous incarnation replays to satisfy and new invocations is really scary.

We've talked about having eager wakers that cause the upgrade to fail if they don't replay successfully to mitigate this, but so far have not implemented this.

Reply:
This option is no longer available except in tests of the replay membrane.